ws-client/src/transport.rs

```diff
@@ -37,6 +37,15 @@ use thiserror::Error;
 
 type TlsOrPlain = crate::stream::EitherStream<TcpStream, TlsStream<TcpStream>>;
 
+#[derive(Clone)]
+pub struct Host(String);
```

Comment:
Introduced a new type to avoid nasty bugs when several params has the same type.

It could be a `pub struct Host<'a>(&'a str);

Reply:
I think `String` is fine.